.github/workflows/test_runner.yaml

```diff
@@ -9,6 +9,7 @@ jobs:
   test_job:
     # If you change this job, please also change it in
     # build_runner.yaml (and see comments there for explanation)
+    if: github.ref_type == 'branch'
```

Comment:
## Suggestion

nit: I'd consider removing this: that is, still running the checks even if it's a tag that's pushed here.

I'd personally probably drop this line to make everything very slightly simpler and more consistent. See below for an explanation of why.

```suggestion
```

## Rationale

1. There's a copy-paste in [`build-runner.yml`](https://github.com/ebmdatalab/datalab-notebook-template/blob/333f5af403965888f908215c210008f25342edaa/.github/workflows/build_runner.yaml#L37-L55) that hasn't yet been changed.
    * We could edit that to match the change here for consistency as the comment here suggests…
    * …but it doesn't really need to be changed there specifically, because the tests would still work anyway: they don't involve pushing to GitHub.
    * And, actually, `test_job` in `build_runner.yaml` won't run anyway for a tagged release. The [`needs`](https://github.com/ebmdatalab/datalab-notebook-template/blob/333f5af403965888f908215c210008f25342edaa/.github/workflows/build_runner.yaml#L48) line requires that `build` job in `build_runner.yaml` successfully completes first before `test_job` in `build_runner.yaml` will start.
2. This `test_runner.yaml` workflow will be run, as far as I can tell, on a tag push. There's no penalty for having this test run on a tag push. It won't fail in the same way that the `build_runner.yaml` currently would break due to trying to push an already existing tag, when that new tag is created and initiates `build_runner.yaml`. [^1]
    * If anything, it means you'll still hopefully get a confirmation that the tests pass when you push the tag. Which lets you know that the tagged version is OK. That :heavy_check_mark: also shows up on the releases page too.

[^1]: For anyone else reading, pushing a tag [bypasses path checks in GitHub Actions](https://github.community/t/why-are-path-filters-not-evaluated-for-pushes-to-tags/181580), as used in the [`build_runner.yaml` workflow](https://github.com/ebmdatalab/datalab-notebook-template/blob/333f5af403965888f908215c210008f25342edaa/.github/workflows/build_runner.yaml#L2-L5).